core/chains/evm/gas/arbitrum_estimator.go

```diff
@@ -28,19 +25,14 @@ type ArbConfig interface {
 	BumpMin() *assets.Wei
 }
 
-//go:generate mockery --quiet --name ethClient --output ./mocks/ --case=underscore --structname ETHClient
-type ethClient interface {
-	CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
-}
-
 // arbitrumEstimator is an Estimator which extends SuggestedPriceEstimator to use getPricesInArbGas() for gas limit estimation.
 type arbitrumEstimator struct {
 	services.StateMachine
 	cfg ArbConfig
 
 	EvmEstimator // *SuggestedPriceEstimator
 
-	client     ethClient
+	client     feeEstimatorClient
```

Comment:
No need to store `client` in a variable. It's being passed inside the suggested estimator and the l1oracle.